core/src/proof.rs

```diff
@@ -100,6 +100,45 @@ pub struct VerifiedPathProof {
     terminal: Option<LeafData>,
 }
 
+impl VerifiedPathProof {
+    /// Get the terminal node.
+    pub fn terminal(&self) -> Option<&LeafData> {
+        self.terminal.as_ref()
+    }
+
+    /// Get the proven path.
+    pub fn path(&self) -> &BitSlice<u8, Msb0> {
+        &self.key_path.view_bits()[..self.depth as usize]
+    }
+
+    /// Check whether this path resolves to the given leaf.
+    ///
+    /// A return value of `true` confirms that the key indeed has no value in the trie.
+    /// A return value of `false` means that either the key provided is out of scope for this path
+    /// or it does not have the expected value.
+    pub fn confirm_value(&self, expected_leaf: &LeafData) -> bool {
+        self.terminal() == Some(expected_leaf)
+    }
+
+    /// Check whether this proves that a key has no value in the trie.
+    ///
+    /// A return value of `true` confirms that the key indeed has no value in the trie.
+    /// A return value of `false` means that either the provided key is out of scope or it has
+    /// a value.
+    pub fn confirm_nonexistence(&self, key_path: &KeyPath) -> bool {
+        let this_path = self.path();
+        let other_path = &key_path.view_bits::<Msb0>()[..self.depth as usize];
```

Comment:
untrusted input requires some care.

Reply:
`depth` is a u8, `key_path` has 256 bits, it's impossible to fail, no?